src/main/java/solution/EnumBoard.java

```diff
@@ -0,0 +1,240 @@
+        SudokuCharacter[] columnCharacters = new SudokuCharacter[GRID_DIMENSION];
+        int iterator = 0;
+
+        for (int column = 0; column < GRID_DIMENSION; column++) {
+            columnCharacters[iterator] = board[iterator][coordinates.y];
+            iterator++;
+        }
+        return columnCharacters;
+    }
+
+    private SudokuCharacter[] getBoxCharacters(Point coordinates) {
+        Point boxCoordinates = getBoxCoordinates(coordinates);
+        SudokuCharacter[] boxCharacters = new SudokuCharacter[GRID_DIMENSION];
+        int iterator = 0;
+
+        for (int row = 0; row < BOX_DIMENSIONS; row++) {
+            for (int column = 0; column < BOX_DIMENSIONS; column++) {
+                boxCharacters[iterator] = board[row + boxCoordinates.x][column + boxCoordinates.y];
+                iterator++;
+            }
+        }
+        return boxCharacters;
+    }
+
+    private Point getBoxCoordinates(Point variableCoordinate) {
+        return new Point(
+                variableCoordinate.x / BOX_DIMENSIONS * BOX_DIMENSIONS,
+                variableCoordinate.y / BOX_DIMENSIONS * BOX_DIMENSIONS
+        );
+    }
+
+    public void writeCharacterTo(SudokuCharacter inputCharacter, Point coordinates) {
+        if (inputCharacter == null) {
+            throw new NullPointerException("Input character cannnot be null.");
+        }
+        if (coordinates == null) {
+            throw new NullPointerException("Coordinates cannnot be null.");
+        }
+        if (!isCoordinatesLegal(coordinates)) {
+            throw new IllegalArgumentException(
+                    "Coordinates [" + coordinates.x + "," + coordinates.y + "] are not writable."
+            );
+        }
+        board[coordinates.x][coordinates.y] = inputCharacter;
+    }
+
+    private boolean isCoordinatesLegal(Point potentialCoordinates) {
+        return allVariableCoordinates.contains(potentialCoordinates);
+    }
+
+    public SudokuCharacter getIncrementedCharacter(Point coordinates) {
+        if (coordinates == null) {
+            throw new NullPointerException("Coordinates cannot be null.");
+        }
+        return SudokuCharacter.getNext(getCharacter(coordinates));
+    }
+
+    public SudokuCharacter getEmptyCharacter() {
+        return SudokuCharacter.getEmptyCharacter();
+    }
+
+    public void printBoard() {
+        System.out.println("-------------------------------");
+        for (int i = 0; i < GRID_DIMENSION; i++) {
+            for (int j = 0; j < GRID_DIMENSION; j++) {
+                if (j % 3 == 0) {
+                    System.out.print('|');
+                }
+                System.out.print(" " + board[i][j].getChar() + " ");
+            }
+            System.out.println("|");
+            if (i % 3 == BOX_DIMENSIONS - 1) {
+                System.out.println("-------------------------------");
+            }
+        }
+    }
+
+    public boolean equals(Object other) {
```

Comment:
Interesting, do you consider comparing two boards or storing boards in any collection?